diff --git a/cmake/config-ix.cmake b/cmake/config-ix.cmake
index 01b30568a..5e35268c7 100644
--- a/cmake/config-ix.cmake
+++ b/cmake/config-ix.cmake
@@ -365,6 +365,8 @@ elseif (LLVM_NATIVE_ARCH MATCHES "wasm32")
   set(LLVM_NATIVE_ARCH WebAssembly)
 elseif (LLVM_NATIVE_ARCH MATCHES "wasm64")
   set(LLVM_NATIVE_ARCH WebAssembly)
+elseif (LLVM_NATIVE_ARCH MATCHES "e2k")
+  set(LLVM_NATIVE_ARCH Elbrus2000)
 elseif (LLVM_NATIVE_ARCH MATCHES "riscv64")
   set(LLVM_NATIVE_ARCH RISCV)
 elseif (LLVM_NATIVE_ARCH MATCHES "e2k")
diff --git a/external/SPIRV-Tools b/external/SPIRV-Tools
--- a/external/SPIRV-Tools
+++ b/external/SPIRV-Tools
@@ -1 +1 @@
-Subproject commit 0cfe9e7219148716dfd30b37f4d21753f098707a
+Subproject commit 0cfe9e7219148716dfd30b37f4d21753f098707a-dirty
